tools/reckless

```diff
@@ -0,0 +1,684 @@
+    except PermissionError:
+        print(f"Permission denied removing dir: {target}")
+    return False
+
+
+class Config():
+    """A generic class for procuring, reading and editing config files"""
+    def obtain_config(self, config_path, default_text, warn=False):
+        """Return a config file from the desired location. Create one with
+        default_text if it cannot be found."""
+        if isinstance(config_path, type(None)):
+            raise Exception("Generic config must be passed a config_path.")
+        assert isinstance(config_path, str)
+        # FIXME: warn if reckless dir exists, but conf not found
+        if os.path.exists(config_path):
+            with open(config_path, 'r+') as f:
+                config_content = f.readlines()
+            return config_content
+        print(f'config file not found: {config_path}')
+        if warn:
+            confirm = input('press [Y] to create one now.\n').upper() == 'Y'
+        else:
+            confirm = True
+        if not confirm:
+            sys.exit(1)
+        parent_path = os.path.split(config_path)[0]
+        # Create up to one parent in the directory tree.
+        if create_dir(1, parent_path):
+            with open(self.conf_fp, 'w') as f:
+                f.write(default_text)
+                # FIXME: Handle write failure
+                return default_text
+        else:
+            print('could not create the parent directory')
+            return None
+
+    def editConfigFile(self, addline, removeline):
+        remove_these_lines = []
+        with open(self.conf_fp, 'r') as reckless_conf:
+            original = reckless_conf.readlines()
+            empty_lines = []
+            for n, l in enumerate(original):
+                if l.strip() == removeline:
+                    remove_these_lines.append(n)
+                    continue
+                if l.strip() == '':
+                    empty_lines.append(n)
+                    if n-1 in empty_lines:
+                        # The white space is getting excessive.
+                        remove_these_lines.append(n)
+                        continue
+            with open(self.conf_fp, 'w') as conf_write:
+                # no need to write if passed 'None'
+                line_exists = not bool(addline)
+                for n, l in enumerate(original):
+                    if n not in remove_these_lines:
+                        if n > 0:
+                            conf_write.write(f'\n{l.strip()}')
+                        else:
+                            conf_write.write(l.strip())
+                        if addline == l:
+                            # addline is idempotent
+                            line_exists = True
+                if not line_exists:
+                    conf_write.write(f'\n{addline}')
+
+    def __init__(self, path=None, default_text=None, warn=False):
+        assert path is not None
```

Comment:
No point in declaring `path=None` and then failing if it is `None`. Default args suggest that you can omit them. Same for `default_text`.